pwa.php

```diff
@@ -132,6 +116,38 @@ function _pwa_incorrect_plugin_slug_admin_notice() {
 /** Hooks to add for when accessing admin. */
 require_once PWA_PLUGIN_DIR . '/wp-admin/admin.php';
 
+/**
+ * Filters whether service worker integrations should be enabled.
+ *
+ * As these are experimental, they are kept separate from the service worker core code and hidden behind a feature flag.
+ *
+ * Instead of using this filter, you can also use a constant `WP_SERVICE_WORKER_INTEGRATIONS_ENABLED`.
+ *
+ * @since 0.2
+ *
+ * @param bool $enabled Whether or not service worker integrations are enabled.
+ */
+if ( apply_filters( 'wp_service_worker_integrations_enabled', defined( 'WP_SERVICE_WORKER_INTEGRATIONS_ENABLED' ) && WP_SERVICE_WORKER_INTEGRATIONS_ENABLED ) ) {
+	/** WP_Service_Worker_Integration Interface */
+	require_once PWA_PLUGIN_DIR . '/wp-includes/integrations/interface-wp-service-worker-integration.php';
```

Comment:
Can we move the entire `integrations` directory out of `wp-includes`? That will make it much clearer what is in scope for a merge proposal and core patch.